test/regression/ifconv.c

```diff
@@ -0,0 +1,129 @@
+#include <stdio.h>
+
+/* Several equivalent forms that should be turned into cmov */
```

Comment:
I was wondering if you could clarify this (for my benefit, this is not actually a PR review comment): from reading this test, my understanding is that it checks whether code involving the relevant pattern (`b ? x : y` and `if b then x else y`) has the correct observable behaviour. As far as I can tell, this test says nothing about when the optimisation implemented in this PR is (or ought to be) triggered - that is, this test should succeed even on a version of CompCert without the optimisation. Did I get that right?

Reply:
Yes, you're entirely right.  To see where the optimization triggered, I look at the generated assembly code.